src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -51,6 +52,42 @@ struct bdf
   }
 };
 
+struct kds_custats
+{
+  using result_type = query::kds_custats::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    std::string errmsg;
+    result_type cuStats;
+    auto pdev = get_pcidev(device);
+  
+    if (!std::getenv("XCL_SKIP_CU_READ")) {
+        try {
+	    // lock xclbin
+            auto dev = const_cast <xrt_core::device *>(device);
+            auto uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(dev));
+            dev->open_context(uuid.get(), -1, true);
```

Comment:
I really don't like "magic" values / variables.  In this case, it would be helpful if "-1" was define to be something more description.